src/irc/core/servers-redirect.c

```diff
@@ -442,7 +442,7 @@ static void redirect_abort(IRC_SERVER_REC *server, REDIRECT_REC *rec)
 		g_free(str);
 
 		if (rec->failure_signal != NULL)
-			signal_emit(rec->failure_signal, 1, server);
+			signal_emit(rec->failure_signal, 3, server, rec->cmd->name, rec->arg);
```

Comment:
Is this safe?

Reply:
It's an API change, we add 2 more arguments to the failure signal. As far as I know existing code is also sloppy on arguments so this should not be more unsafe than what we have in other places.

(In general the whole signal emission system is a bit unsafe)